internal/dag/httpproxy_processor.go

```diff
@@ -578,9 +578,23 @@ func (p *HTTPProxyProcessor) computeRoutes(
 			}
 
 			var uv *PeerValidationContext
-			if protocol == "tls" || protocol == "h2" {
+			if (protocol == "tls" || protocol == "h2") && service.UpstreamValidation != nil {
+				// If the CACertificate name in the UpstreamValidation is namespaced and the namespace
+				// is not the proxy's namespace, check if the referenced secret is permitted to be
+				// delegated to the proxy's namespace.
+				// By default, a non-namespaced CACertificate is expected to reside in the proxy's namespace.
+				caCertNamespacedName, err := getNamespacedName(service.UpstreamValidation.CACertificate)
+				if err != nil {
+					// The CACertificate name is not namespaced, scope it be in the proxy's namespace.
+					caCertNamespacedName = types.NamespacedName{Name: service.UpstreamValidation.CACertificate, Namespace: proxy.Namespace}
+				}
+				if caCertNamespacedName.Namespace != proxy.Namespace && !p.source.DelegationPermitted(caCertNamespacedName, proxy.Namespace) {
```

Comment:
Looks like `DelegationPermitted(...)` properly handles the case where the namespaces are equivalent, so no need to explicitly check that here

Applies to the extension processor too.